chain/client/src/sync/state.rs

```diff
@@ -535,61 +520,33 @@ impl StateSync {
     /// Only select candidates that we have no pending request currently ongoing.
     fn possible_targets(
         &mut self,
-        me: &Option<AccountId>,
         shard_id: ShardId,
-        chain: &Chain,
-        epoch_manager: &dyn EpochManagerAdapter,
-        sync_hash: CryptoHash,
         highest_height_peers: &[HighestHeightPeerInfo],
-    ) -> Result<Vec<AccountOrPeerIdOrHash>, near_chain::Error> {
-        let prev_block_hash = *chain.get_block_header(&sync_hash)?.prev_hash();
-        let epoch_hash = epoch_manager.get_epoch_id_from_prev_block(&prev_block_hash)?;
-
-        let block_producers =
-            epoch_manager.get_epoch_block_producers_ordered(&epoch_hash, &sync_hash)?;
-        let peers = block_producers
+    ) -> Result<Vec<PeerId>, near_chain::Error> {
+        let peers = highest_height_peers
```

Comment:
You no longer consider the validators as candidates, do you? Are there any drawbacks to this approach? Do we lose the guarantee that at least some peers will for sure have the needed data?

Reply:
I argue that mainnet validators block requests related to state sync and therefore removing them has no effect on state sync, or even makes it more reliable.
Given the low diameter of the network topology graph, such a peer should be available in TIER2. And if not, then that should be handled by the decentralized state sync separately.

Reply:
The current implementation does not work and the new design will talk to peers regardless of their status in the network.

Reply:
> I argue that mainnet validators block requests related to state sync and therefore removing them has no effect on state sync, or even makes it more reliable. 

Thank you, that makes sense. 

> Given the low diameter of the network topology graph, such a peer should be available in TIER2.  

I believe all peers are available in TIER2. I assumed that we're talking about TIER2 direct peers here, is that right? 

> And if not, then that should be handled by the decentralized state sync separately. 

Yep, that makes perfect sense to me. I'm not sure where exactly stateless validation is heading in terms of number of shards but let's design the solution to handle a case where it's quite likely none of the direct peers have the required state. This can happen e.g. when #shards > #direct peers.